src/main/java/org/openrewrite/java/migrate/lang/var/UseVarForGenericsConstructors.java

```diff
@@ -145,108 +135,36 @@ private static boolean hasBounds(JavaType type) {
-                return new J.Primitive(Tree.randomId(), Space.EMPTY, Markers.EMPTY, primitiveType);
-            }
-            if (type instanceof JavaType.Class) {
-                String className = ((JavaType.Class) type).getClassName();
-                return new J.Identifier(Tree.randomId(), Space.EMPTY, Markers.EMPTY, emptyList(), className, type, null);
-            }
-            if (type instanceof JavaType.Array) {
-                TypeTree elemType = (TypeTree) typeToExpression(((JavaType.Array) type).getElemType());
-                return new J.ArrayType(Tree.randomId(), Space.EMPTY, Markers.EMPTY, elemType, null, JLeftPadded.build(Space.EMPTY), type);
-            }
-            if (type instanceof JavaType.GenericTypeVariable) {
-                String variableName = ((JavaType.GenericTypeVariable) type).getName();
-                J.Identifier identifier = new J.Identifier(Tree.randomId(), Space.EMPTY, Markers.EMPTY, emptyList(), variableName, type, null);
-
-                List<JavaType> bounds1 = ((JavaType.GenericTypeVariable) type).getBounds();
-                if (bounds1.isEmpty()) {
-                    return identifier;
-                } else {
-                    /*
-                    List<JRightPadded<TypeTree>> bounds = bounds1.stream()
-                            .map(b -> new J.Identifier(Tree.randomId(), Space.EMPTY, Markers.EMPTY, , null, null))
-                            .map(JRightPadded::build)
-                            .collect(Collectors.toList());
-
-                    return new J.TypeParameter(Tree.randomId(), Space.EMPTY, Markers.EMPTY, new ArrayList<>(), identifier, JContainer.build(bounds));
-                     */
-                    throw new IllegalStateException("Generic type variables with bound are not supported, yet.");
-                }
-            }
-            if (type instanceof JavaType.Parameterized) { // recursively parse
-                List<JavaType> typeParameters = ((JavaType.Parameterized) type).getTypeParameters();
-
-                List<JRightPadded<Expression>> typeParamsExpression = new ArrayList<>(typeParameters.size());
-                for (JavaType curType : typeParameters) {
-                    typeParamsExpression.add(JRightPadded.build(typeToExpression(curType)));
-                }
-
-                NameTree clazz = new J.Identifier(Tree.randomId(), Space.EMPTY, Markers.EMPTY, emptyList(), ((JavaType.Parameterized) type).getClassName(), null, null);
-                return new J.ParameterizedType(Tree.randomId(), Space.EMPTY, Markers.EMPTY, clazz, JContainer.build(typeParamsExpression), type);
-            }
-
-            throw new IllegalArgumentException(String.format("Unable to parse expression from JavaType %s", type));
+            J.NewClass initializer = Objects.requireNonNull((J.NewClass) vd.getVariables().get(0).getInitializer());
```

Comment:
Do we have evidence that the other recipes around local variable type inference are effected by this as well? If so we should align here